generated/src/aws-cpp-sdk-s3-crt/source/S3CrtClient.cpp

```diff
@@ -769,9 +772,9 @@ void S3CrtClient::CopyObjectAsync(const CopyObjectRequest& request, const CopyOb
   aws_s3_meta_request_options options;
   AWS_ZERO_STRUCT(options);
   aws_uri endpoint;
-  AWS_ZERO_STRUCT(endpoint);
+  InitCrtEndpointFromUri(&endpoint, endpointResolutionOutcome.GetResult().GetURI());
   options.endpoint = &endpoint;
-  std::unique_ptr<aws_uri, void(*)(aws_uri*)> endpointCleanup { options.endpoint, &aws_uri_clean_up };
+  std::unique_ptr<aws_uri, void(*)(aws_uri*)> endpointCleanup { &endpoint, &aws_uri_clean_up };
```

Comment:
It looks okay but I simply don't like this (initial design)
usage of a unique pointer to wrap some object on stack by getting the pointer......... some small refactoring of this function re-arranging the order of variables on stack and boom!